### DISCUSSION TITLE
Updating code for version 0.20 - channel_display_groups

### DISCUSSION THREAD

**Discussion by NoaMills-USDA** [azavea/raster-vision#1656] [Q&A] [answered]
I have code for a semantic segmentation model that worked in version 0.13. I'm trying to create a new version of the same code that's compatible with rastervision 0.20. When I first ran my original (0.13) code in version 0.20, I ran into an error stating that the field "default_class_id" is not permitted in GeoJSONVectorSourceConfig. Through the documentation and the new quickstart code, I was able to figure out how to fix this by passing the "transformers" list. instead of the "default_class_id". That is the only change I have made so far, and the rest of my code is written for version 0.13. I tried running it again and ran into a similar error: the "channel_display_groups" field is not permitted in SemanticSegmentationGeoDataConfig. I have two questions:

1. How do I specify the channel_display_groups information for SemanticSegmentationGeoDataConfig in version 0.20?
2. Is there a more systematic way for me to update my code to version 0.20 other than running my previous code and tackling bugs when they come up?

**Comment by AdeelH** [comment, marked as answer]
Apologies for the late response.

1. `channel_display_groups` is now supposed to go in `plot_options` as shown [here](https://github.com/azavea/raster-vision/blob/master/rastervision_pytorch_backend/rastervision/pytorch_backend/examples/semantic_segmentation/isprs_potsdam.py#L188-L190). 
2. ~We do not have such a checklist yet. But it would definitely be nice to have. One thing you can do is to take a look at the [diff between the two branches](https://github.com/azavea/raster-vision/compare/0.13...0.20). Go to the "Files Changed" tab, scroll down to the example files (`isprs_potsdam.py` etc.), and see what's changed.~ Here is a non-exhaustive list of required changes: https://github.com/azavea/raster-vision/discussions/1666.
